typed/typed.go

```diff
@@ -59,13 +59,35 @@ func (tv TypedValue) Validate() error {
 	return errs
 }
 
+// ToFieldSet creates a set containing every leaf field mentioned in tv, plus
+// any validation errors. A (perhaps partial) field set is returned even if
+// there are validation errors.
+func (tv TypedValue) ToFieldSet() (*fieldpath.Set, error) {
```

Comment:
Are the two methods different enough that we don't want to merge them? AFAIU, `err := tv.Validate()` is equivalent to `_, err := tv.ToFieldSet()`?

Reply:
I think it's good to keep the interfaces separate, so we know the caller's intent if we ever change the library.

Reply:
I don't disagree with that, you shouldn't have to copy the implementation though:
```
func (tv) Validate() err {
  _, err := tv.ToFieldSet()
  return err
}
```

Reply:
There's some modicum of overhead to constructing a field set. I suppose I could make a constructor to reduce the duplication.

Reply:
Fixed